src/plcc/load_spec/validate_spec/validate_syntactic_spec/resolve_repeating_rules_test.py

```diff
@@ -0,0 +1,136 @@
+from ...parse_spec.parse_syntactic_spec import (
+    SyntacticSpec,
+    Symbol,
+    LhsNonTerminal,
+    Terminal,
+)
+from .resolve_repeating_rules import resolve_repeating_rules
+
+
+def test_standard_unaffected():
+    rule_1 = makeStandardSyntacticRule(
+        makeLine("<sentence> ::= VERB"),
+        makeLhsNonTerminal("sentence"),
+        [makeTerminal("VERB")],
+    )
+    rule_2 = makeStandardSyntacticRule(
+        makeLine("<name> ::= WORD"),
+        makeLhsNonTerminal("name"),
+        [makeTerminal("WORD")],
+    )
+
+    oldSpec = [rule_1, rule_2]
+    newSpec = resolve_repeating_rules(SyntacticSpec(oldSpec))
+    assert oldSpec == newSpec
+
+
+def test_no_separator():
+    rule_1 = makeStandardSyntacticRule(
+        makeLine("<sentence> ::= VERB"),
+        makeLhsNonTerminal("sentence"),
+        [makeTerminal("VERB")],
+    )
+    rule_2 = makeRepeatingSyntacticRule(
+        makeLine("<name> **= WORD"),
+        makeLhsNonTerminal("name"),
+        [makeTerminal("WORD")],
+    )
+
+    expectedSpec = [
+        rule_1,
+        makeStandardSyntacticRule(
+            makeLine("<name> ::= WORD <name>"),
+            makeLhsNonTerminal("name"),
+            [makeTerminal("WORD"), makeRhsNonTerminal("name")],
+        ),
+        makeStandardSyntacticRule(
+            makeLine("<name>:void ::="), makeLhsNonTerminal("name", "void"), []
+        ),
+    ]
+
+    oldSpec = [rule_1, rule_2]
+    resolvedSpec = resolve_repeating_rules(SyntacticSpec(oldSpec))
+    assert resolvedSpec == expectedSpec
+
+
+def test_with_separator():
+    rule_1 = makeStandardSyntacticRule(
+        makeLine("<sentence> ::= VERB"),
+        makeLhsNonTerminal("sentence"),
+        [makeTerminal("VERB")],
+    )
+    rule_2 = makeRepeatingSyntacticRule(
+        makeLine("<name> **= WORD +SEP"),
+        makeLhsNonTerminal("name"),
+        [makeTerminal("WORD")],
+        separator=Terminal("SEP"),
+    )
+
+    expectedSpec = [
+        rule_1,
+        makeStandardSyntacticRule(
+            makeLine("<name> ::= WORD <name#>"),
+            makeLhsNonTerminal("name"),
+            [makeTerminal("WORD"), makeRhsNonTerminal("name#")],
+        ),
+        makeStandardSyntacticRule(
+            makeLine(f"<name>:void ::="), makeLhsNonTerminal("name", "void"), []
+        ),
+        makeStandardSyntacticRule(
+            makeLine("<name#>:void ::= SEP WORD name#"),
+            makeLhsNonTerminal("name#", "void"),
+            [makeTerminal("WORD"), makeRhsNonTerminal("name#")],
+        ),
+        makeStandardSyntacticRule(
+            makeLine("<name#>:void ::="), makeLhsNonTerminal("name#", "void"), []
+        ),
+    ]
+
+    oldSpec = [rule_1, rule_2]
+    resolvedSpec = resolve_repeating_rules(SyntacticSpec(oldSpec))
+    assert resolvedSpec == expectedSpec
+
+def makeSyntacticSpec(ruleList=None):
+    return SyntacticSpec(ruleList)
+
+
+def makeStandardSyntacticRule(line: Line, lhs: LhsNonTerminal, rhsList: List[Symbol]):
```

Comment:
Could the line parameter be eliminated by constructing the line from the others? This would remove a lot of duplication from the tests.

Similarly, since the lhs parameter is always an LhsNonTerminal, you could accept a str for this parameter and construct the LhsNonTerminal here. This would again simplify the tests.

Reply:
Similar suggestion for `makeRepeating*` and `makeSeparared*`.